org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/TextDocument.java

```diff
@@ -179,7 +179,7 @@ public void update(List<TextDocumentContentChangeEvent> changes) {
 						int length = 0;
 
 						if (range != null) {
-							length = changeEvent.getRangeLength().intValue();
+							length = offsetAt(range.getEnd()) - offsetAt(range.getStart());

```

Comment:
length  = changeEvent.getRangeLength() != null ? changeEvent.getRangeLength().intValue() : offsetAt .....

To continue to support deprecated rangeLength